docs/API.md

```diff
@@ -1,681 +1,1085 @@
+
+## API method: `get_test_history`
 
- -  params: an object containing the following parameters
-    -  frontend\_params: the usual structure containing all the
-       parameters of the interface
-    -  offset: the start of pagination (not yet supported) (optional, default 0)
-    -  limit: number of items to return (not yet supported) (optional, default 200)
- -  jsonrpc: « 2.0 »
- -  id: any kind of unique id allowing to match requests and responses
- -   method: the name of the called method
+Returns a list of completed *tests* for a domain.
 
-**Response**:
+Example request:
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 7,
+  "method": "get_test_history",
+  "params": {
+    "offset": 0,
+    "limit": 200,
+    "frontend_params": {
+      "client_id": "Zonemaster Dancer Frontend",
+      "domain": "zonemaster.net",
+      "profile": "default_profile",
+      "client_version": "1.0.1",
+      "nameservers": [
+        {
+          "ns": "ns3.nic.se",
+          "ip": "2001:67c:124c:2007::45"
+        },
+        {
+          "ns": "ns2.nic.fr",
+          "ip": "192.93.0.4"
+        }
+      ],
+      "ds_info": [],
+      "advanced": true,
+      "ipv6": true,
+      "ipv4": true
+    }
+  }
+}
 ```
+
+Example response:
+```json
 {
+  "id": 7,
   "jsonrpc": "2.0",
-  "id": 140743003648550,
   "result": [
     {
-      "advanced_options": "1",
-      "id": 3,
-      "creation_time": "2014-08-05 19:41:14.522656",
-      "overall_result" : "error"
+      "id": "c45a3f8256c4a155",
+      "creation_time": "2016-11-15 11:53:13.965982",
+      "overall_result": "error",
+      "advanced_options": null
     },
     {
-      "advanced_options": "1",
-      "id": 1,
-      "creation_time": "2014-08-05 11:48:18.542216",
-      "overall_result" : "warning"
-    }
+      "id": "32dd4bc0582b6bf9",
+      "creation_time": "2016-11-14 08:46:41.532047",
+      "overall_result": "error",
+      "advanced_options": null
+    },
+    ...
   ]
 }
 ```
 
- -  jsonrpc: « 2.0 »
- -  id: any kind of unique id allowing to match requests and responses
- -  result: an ordered (starting by the most recent test) list of tests
-    with
-    -  id: the id to use to retrieve the test result
-    -  creation\_date: the date of test
-    -  advanced\_options: if set to 1 serves to differentiate tests
-       with advanced options from those without this option.
-    - overall\_result: shows if there were any errors or warnings in the result (for color differentiation in the test results history)
+>
+> Note: The above example response was abbreviated for brevity to only include
+> the first two elments in each list. Omitted elements are denoted by a `...`
+> symbol.
+>
 
-## Batch mode API (Experimental as of now)
 
-### *JSON-RPC Call*: `add_api_user`
+#### `"params"`
 
-**Request**:
-```
+An object with the following properties:
+
+* `"offset"`: An integer, optional. (default: 0).
+* `"limit"`: An integer, optional. (default: 200).
+* `"frontend_params"`: As described below.
```

Comment:
@mtoma, what is the status of "offset" and "limit"? I can see code for Mysql and PostgreSQL, but is it working?